### ISSUE TITLE
`System.ArgumentOutOfRangeException`: Maximum range is less than the minimum range.

### ISSUE THREAD

**Issue by matjaeck** [OpenRA/OpenRA#16172]
In 4v4 bot only teamgame on bleed, fastest speed.

```md
OpenRA engine version {DEV_VERSION}
Red Alert mod version {DEV_VERSION}
on map 20dc4c16f55d2d2ff040849b1d5e7b8b2efb5322 (LAB-Divide-NoF 5.5 by Community Effort - Script by a bored Coder).
Date: 2019-02-06 00:51:32Z
Operating System: Linux (Unix 4.15.0.45)
Runtime Version: Mono 5.18.0.240 (tarball Wed Jan 16 09:10:16 UTC 2019) CLR 4.0.30319.42000
Exception of type `System.ArgumentOutOfRangeException`: Maximum range is less than the minimum range.
Parameter name: maxRange
  at OpenRA.Map+<FindTilesInAnnulus>c__Iterator0.MoveNext () [0x00033] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at System.Linq.Enumerable+WhereEnumerableIterator`1[TSource].MoveNext () [0x0004e] in <0a52baf64a2a4f9b96857c7b358eb12d>:0 
  at OpenRA.Mods.Common.Activities.MoveAdjacentTo.CalculatePathToTarget (OpenRA.Actor self) [0x00068] in <3fd50f4779394c92bce7f61c1a7f9a75>:0 
  at OpenRA.Mods.Common.Activities.MoveAdjacentTo+<Tick>c__AnonStorey0.<>m__0 () [0x00000] in <3fd50f4779394c92bce7f61c1a7f9a75>:0 
  at OpenRA.Mods.Common.Activities.Move.EvalPath () [0x00001] in <3fd50f4779394c92bce7f61c1a7f9a75>:0 
  at OpenRA.Mods.Common.Activities.Move.Tick (OpenRA.Actor self) [0x000be] in <3fd50f4779394c92bce7f61c1a7f9a75>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x0005f] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00022] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Mods.Common.Activities.MoveAdjacentTo.Tick (OpenRA.Actor self) [0x00214] in <3fd50f4779394c92bce7f61c1a7f9a75>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x0005f] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00022] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Actor.Tick () [0x00010] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.World.Tick () [0x00046] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00237] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Game.LogicTick () [0x00050] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Game.Loop () [0x000d6] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Game.Run () [0x00042] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00011] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
  at OpenRA.Program.Main (System.String[] args) [0x0004f] in <bd28f51dd31e43d4ab46b85c35b1de1a>:0 
```

Map has custom rules but they only affect prerequisites so I don't think it is related.

**Comment by pchote** [comment]
Can you please upload the replay for this match and note which bleed commit this was on?

**Comment by matjaeck** [comment]
Crashes also in [Replay](https://drive.google.com/open?id=1gue9CQxwNM9nbg7cZ6doNMHaFdsYeyKR)

Played on 297be6d6cc70b7ffd6ad544211c8020f510e4248 I think.

**Comment by pchote** [comment]
This is caused by a cruiser trying to target a missile sub. `Weapon.IsValidAgainst` is returning false for all armaments, which causes `AttackFollow.lastVisibleMaximumRange` to be set to `WRange.Zero`. It loses sight of the sub, probably because it submerges, and then tries to `MoveWithinRange` with a `minRange` of 3 cells and a `maxRange` of 0.  This kills the OpenRA.

**Comment by pchote** [comment]
Dropping the target in this case is probably the best result.

**Comment by matjaeck** [comment]
Easiest way to reproduce is to emerge with two subs of player A within vision* of a cruiser of player B, let the cruiser auto-target and kill the first sub and then target the other sub manually. When it submerges, the game crashes. This can also work with one sub but it is hard to avoid the first shot of the cruiser and it looks like this only happens on the second, manually issued attack order.

*not sure if detection range has anything to do with it, see below


<details>
  <summary>GIF</summary>

![peek 2019-02-07 16-04](https://user-images.githubusercontent.com/37185497/52420369-43d51100-2af2-11e9-83e9-c58d0bdc6b27.gif)

</details>

**Comment by GraionDilach** [comment]
The proposed soultion sounds like a possible fix to #10908 as well.

**Comment by rautamiekka** [comment]
Why is it even possible to set the max below minimum ? Shouldn't such a protection be included in the datatype or variable itself, custom-designed by 'OpenRA' ?

**Comment by GraionDilach** [comment]
For the latter part - there are a huge amounts of discussion within the community scattered across multiple tickets , forums, and Discord channels that players do expect the targets being tracked incase the attacker follows them and reveals them that way.

**Comment by obrakmann** [comment]
> The proposed solution sounds like a possible fix to #10908 as well.

The stacktraces are very similar, yes. I guess we will close that as well.
